frontend/catalyst/passes.py

```diff
@@ -371,34 +365,37 @@ def circuit(x: float):
     >>> circuit(0.54)
     Array(0.5965506257017892, dtype=float64)
     """
-    if not isinstance(fn, qml.QNode):
-        raise TypeError(f"A QNode is expected, got the classical function {fn}")
+    if not isinstance(qnode, qml.QNode):
+        raise TypeError(f"A QNode is expected, got the classical function {qnode}")
 
-    funcname = fn.__name__
-    wrapped_qnode_function = fn.func
+    clone = copy.copy(qnode)
+    clone.__name__ += "_merge_rotations"
 
-    def wrapper(*args, **kwrags):
-        if EvaluationContext.is_tracing():
-            apply_registered_pass_p.bind(
-                pass_name="merge-rotations",
-            )
-        return wrapped_qnode_function(*args, **kwrags)
+    @functools.wraps(clone)
+    def wrapper(*args, **kwargs):
+        pass_pipeline = kwargs.pop("pass_pipeline", tuple())
+        pass_pipeline += (Pass("merge-rotations"),)
+        kwargs["pass_pipeline"] = pass_pipeline
+        return clone(*args, **kwargs)
 
-    fn_clone = copy.copy(fn)
-    fn_clone.func = wrapper
-    fn_clone.__name__ = funcname + "_merge_rotations"
-
-    return fn_clone
+    return wrapper
 
 
 def _API_name_to_pass_name():
     return {"cancel_inverses": "remove-chained-self-inverse", "merge_rotations": "merge-rotations"}
```

Comment:
Should `{"ions_decomposition": "ions-decomposition"}` be added to this dictionary as well?

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1349/commits/187c8bd394c320a142c46a6651c2e251bef1adf8